pkg/deploy/lattice/target_group_synthesizer_test.go

```diff
@@ -371,14 +374,29 @@ func Test_SynthesizeSDKTargetGroups(t *testing.T) {
 			wantDataStoreError:   nil,
 			wantDataStoreStatus:  "",
 		},
+		{
+			name: "Delete SDK TargetGroup since it is dangling with no service associated",
```

Comment:
When would this case happen under normal circumstances? Want to make sure we're not being over eager with our deletes.

Reply:
This happens when changing target group protocol - it should create `tg-https-http1` and remove `tg-http-http1` for example. Due to how we have detected `inUse` target group before, I need to make a change in behavior here (also related to the comment above)